src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -791,4 +888,84 @@ public void UseProxy_GetSet_Roundtrips()
             }
         }
     }
+
+    public sealed class SocketsHttpHandler_ExternalConfiguration_Test : HttpClientTestBase
+    {
+        private const string EnvironmentVariableSettingName = "DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER";
+        private const string AppContextSettingName = "System.Net.Http.UseSocketsHttpHandler";
```

Comment:
fyi.  Naming conventions for AppContextSwitches are that they usually start with "Switch.".

See: 
https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element
https://msdn.microsoft.com/en-us/library/system.appcontext(v=vs.110).aspx

Reply:
@eerhardt, @AlexGhiondea, what naming should be used here?  I'm not introducing the name in this PR and so not going to block the PR on it, but we can change it subsequently if this should be something different.

Reply:
Starting with `Switch.` isn't the naming convention we've been using in other .NET Core app settings.  Ex:

```
      "System.GC.Server": true,
      "System.GC.Concurrent": true,
      "System.GC.RetainVM": true,
      "System.Threading.ThreadPool.MinThreads": 4,
      "System.Threading.ThreadPool.MaxThreads": 25,
      "System.Globalization.Invariant": true
```

/cc @tarekgh

Reply:
CC @AlexGhiondea

Reply:
Not sure why .NET Core started naming AppContext switches without the "Switch." prefix in their name. But .NET Framework switches have used "Switch." everywhere.

Reply:
@davidsh this is true for the environment variables too. we use different prefix there too. so, in general, we have different conventions between the net core and the full framework.

Reply:
>in general, we have different conventions between the net core and the full framework.

Was that intended?  I don't see the benefit of having it different between .NET Core and .NET Framework.

Reply:
at least for the environment variables, naming is intended. we need to differentiate the full framework variables from the full framework ones. full framework one wouldn't make sense in net core (having the names prefixed with COMPLUS_).